### PR TITLE
Remove list.c

### LINKED ISSUE #2: Remove list.c
This will require some refactoring. Can probably replace it with wl_list.

### PR DESCRIPTION
**Pull request by RedSoxFan**
Closes #2 

This replaces all occurrences of `list_t` with `wl_list` to eliminate
the need for `list.c`

### CODE REVIEW

File: loop.c [hunk 1]
```diff
@@ -145,34 +157,37 @@ struct loop_timer *loop_add_timer(struct loop *loop, int ms,
...
+bool loop_remove_timer(struct loop *loop, struct loop_timer *remove) {
+	struct loop_timer *timer = NULL, *tmp_timer = NULL;
+	wl_list_for_each_safe(timer, tmp_timer, &loop->timers, link) {
+		if (timer == remove) {
+			wl_list_remove(&timer->link);
 			free(timer);
 			return true;
 		}
-	}
+	};
```
Comment (emersion) [inline loop.c]: Style: extra `;` here

Comment (RedSoxFan) [reply loop.c]: Fixed... I'm pretty sure gcc was giving me errors about missing semicolons and that's why I added them, but appears to be working now so whatever...

**Comment by emersion** [comment]
Can you rebase?

**Comment by RedSoxFan** [comment]
> Can you rebase?

Done

**Pull request merged**